### PR TITLE
TYP: some typing in internals/concat.py

### PR DESCRIPTION
**Pull request by topper-123**
Some typing.

### CODE REVIEW

File: pandas/core/internals/concat.py [hunk 1]
```diff
@@ -279,7 +282,8 @@ def _concat_managers_axis0(
                 # by slicing instead of copy(deep=False), we get a new array
                 #  object, see test_concat_copy
                 nb = blk.getitem_block(slice(None))
-            nb._mgr_locs = nb._mgr_locs.add(offset)
+            # "BlockPlacement" has no attribute "add"; maybe "append"?
```
Comment (jbrockmendel) [inline pandas/core/internals/concat.py:285]: could add this in _libs.interrnals.pyi

Comment (topper-123) [reply pandas/core/internals/concat.py]: Yeah thanks, I’ve updated.

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Comment by jbrockmendel** [comment]
thanks @topper-123

**Pull request merged**